includes/discovery/vlans/vlan_functions.inc.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/* idea from http://php.net/manual/en/function.hex2bin.php comments */
+function hex2bin_compat($str)
+{
+    if (strlen($str) % 2 !== 0) {
+        trigger_error(__FUNCTION__.'(): Hexadecimal input string must have an even length', E_USER_WARNING);
```

Comment:
Why run trigger_error()? If we have something to tell the user please use d_echo().